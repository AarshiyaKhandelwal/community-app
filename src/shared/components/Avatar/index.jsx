/**
 * Avatar
 *
 * Description:
 *   Reusable user avatar. Depending on wether or not an image URL is
 *   provided, it will display the user's profile picture, or a default
 *   user avatar.
 *
 * Styling:
 *  By default the avatar is round-shaped. Custom styling may be applied using
 *  react-css-themr.
 *
 * Usage:
 *   <Avatar url={imgUrl} theme={customeTheme} />
 *
 * Props:
 *    - theme: Object, optional. Parent-provided theme.
 *    - url: String, optional. Image URL.
 */

import PT from 'prop-types';
import React from 'react';
import { themr } from 'react-css-themr';
import DefaultAvatar from '../../../assets/images/ico-user-default.svg';
import defaultStyle from './style.scss';

function Avatar({ theme, url }) {
  return url
    ? <img alt="Avatar" src={url} className={theme.avatar} />
    : <DefaultAvatar className={theme.avatar} />;
}

Avatar.defaultProps = {
  customStyles: null,
  theme: {
    avatar: '',
  },
  url: null,
};

Avatar.propTypes = {
  theme: PT.shape({
    avatar: PT.string.isRequired,
  }),
  url: PT.string,
};

export default themr('Avatar', defaultStyle)(Avatar);
