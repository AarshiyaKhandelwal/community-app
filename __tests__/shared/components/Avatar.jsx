import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Avatar from 'components/Avatar';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <Avatar />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar url={'some-fake-string'} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar
      theme={{
        avatar: 'test-style',
      }}
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar
      theme={{
        avatar: 'test-style',
      }}
      url={'some-fake-string'}
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
