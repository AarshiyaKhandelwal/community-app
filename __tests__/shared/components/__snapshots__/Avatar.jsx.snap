// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<Avatar
  customStyles={null}
  theme={
    Object {
      "avatar": "",
    }
  }
  url={null}
/>
`;

exports[`Matches shallow shapshot 2`] = `
<Avatar
  customStyles={null}
  theme={
    Object {
      "avatar": "",
    }
  }
  url="some-fake-string"
/>
`;

exports[`Matches shallow shapshot 3`] = `
<Avatar
  customStyles={null}
  theme={
    Object {
      "avatar": "test-style",
    }
  }
  url={null}
/>
`;

exports[`Matches shallow shapshot 4`] = `
<Avatar
  customStyles={null}
  theme={
    Object {
      "avatar": "test-style",
    }
  }
  url="some-fake-string"
/>
`;
